conservator-framework/src/ConservatorFramework.cpp

```diff
@@ -35,8 +43,14 @@ ConservatorFramework::ConservatorFramework(string connectString, int timeout, cl
 }
 
 void ConservatorFramework::start() {
-    //TODO - add some blocking until zoo_state == CONNECTED and throw exception if not connected within timeout
-    this->zk = zookeeper_init(connectString.c_str(), watcher, timeout, cid, 0, 0);
+    this->zk = zookeeper_init(connectString.c_str(), watcher, timeout, cid, (void *) this, 0);
+
+    unique_lock<std::mutex> connectLock(connectMutex);
+    while(!connected) {
+        if(connectCondition.wait_for(connectLock, chrono::seconds(15)) == cv_status::timeout) {
+            throw "timed out waiting to connect to zookeeper";
```

Comment:
Looks good.

After reading up on `wait_for` some more, looks like it can take a Predicate that could replace the `while(!connected) ` check in case of spurious wakeup.